scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GoalBuilder.java

```diff
@@ -173,17 +177,29 @@ private static Expression<Spans> spansOfConstraintExpression(
       final SchedulingDSL.ConstraintExpression constraintExpression) {
     if (constraintExpression instanceof SchedulingDSL.ConstraintExpression.ActivityExpression c) {
       return new ForEachActivitySpans(
-          ($, simResults, environment) -> {
-            final var startTime = $.interval.start;
-            if (!$.type.equals(c.type())) return false;
-            for (final var arg: c
-                .arguments()
-                .map(expr -> expr.evaluateMap(simResults, startTime, environment))
-                .orElse(Map.of())
-                .entrySet()) {
-              if (!arg.getValue().equals($.parameters.get(arg.getKey()))) return false;
+          new TriFunction<>() {
+            @Override
+            public Boolean apply(
+                final ActivityInstance activityInstance,
+                final SimulationResults simResults,
+                final EvaluationEnvironment environment)
+            {
+              final var startTime = activityInstance.interval.start;
+              if (!activityInstance.type.equals(c.type())) return false;
+              for (final var arg : c
+                  .arguments()
+                  .map(expr -> expr.evaluateMap(simResults, startTime, environment))
+                  .orElse(Map.of())
+                  .entrySet()) {
+                if (!arg.getValue().equals(activityInstance.parameters.get(arg.getKey()))) return false;
+              }
+              return true;
+            }
+
+            @Override
+            public String toString() {
+              return "(filter by ActivityExpression)";
```

Comment:
When is this `toString` called?

Reply:
by the coexistence goal default name setting method